### PR TITLE
feat: support route wise page titles

### PR DESCRIPTION
**Pull request by arjunlalb**
## Description
Support route wise page titles

### Testing
Tested manually. UT added.

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/common/src/navigation/navigation.service.ts [hunk 1]
```diff
@@ -34,11 +35,20 @@ export class NavigationService {
...
+    @Inject(APP_TITLE) private readonly appTitle: string
   ) {
     this.event$(RoutesRecognized)
       .pipe(skip(1), take(1))
       .subscribe(() => (this.isFirstNavigation = false));
+
+    this.navigation$
+      .pipe(
+        switchMap(activatedRoute => this.getLeafRoute(activatedRoute).data),
+        map(routeData => this.titleService.setTitle(routeData.title ? `${this.appTitle} | ${routeData.title}` : this.appTitle))
```
Comment (aaron-steinfeld) [inline projects/common/src/navigation/navigation.service.ts:49]: nit: tap

File: projects/common/src/navigation/navigation.service.ts [hunk 2]
```diff
@@ -75,6 +85,10 @@ export class NavigationService {
     return this.currentParamMap.getAll(parameterName);
   }
 
+  private getLeafRoute(activatedRoute: ActivatedRoute): ActivatedRoute {
```
Comment (aaron-steinfeld) [inline projects/common/src/navigation/navigation.service.ts:88]: this is just an alternative impl to `getCurrentActivatedRoute` - don't need both.

File: src/environments/environment.prod.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export const environment = {
   production: true,
-  graphqlUri: '/graphql'
+  graphqlUri: '/graphql',
+  appTitle: 'Hypertrace'
```
Comment (aaron-steinfeld) [inline src/environments/environment.prod.ts:4]: if this is the source of truth now (which makes sense), let's remove the setting in index.html

File: projects/common/src/navigation/navigation.service.ts (hunk 1 above)
Comment (arjunlalb) [reply projects/common/src/navigation/navigation.service.ts]: done

File: projects/common/src/navigation/navigation.service.ts (hunk 2 above)
Comment (arjunlalb) [reply projects/common/src/navigation/navigation.service.ts]: fixed

File: src/environments/environment.prod.ts (hunk 3 above)
Comment (arjunlalb) [reply src/environments/environment.prod.ts:4]: done @aaron-steinfeld

**Review by aaron-steinfeld — APPROVED** [review]

**Review by aaron-steinfeld — DISMISSED** [review]

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**